airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -0,0 +1,461 @@
+        :type verbose: bool
+
+        :return: A List of nodegroup names within the given cluster.
+        :rtype: List
+        """
+        eks_client = self.conn
+        api_call = partial(eks_client.list_nodegroups, clusterName=clusterName)
+
+        return self._list_all(api_call=api_call, response_key="nodegroups", verbose=verbose)
+
+    def _list_all(self, api_call: Callable, response_key: str, verbose: bool) -> List:
+        """
+        Repeatedly calls a provided boto3 API Callable and collates the responses into a List.
+
+        :param api_call: The api command to execute.
+        :type api_call: Callable
+        :param response_key: Which dict key to collect into the final list.
+        :type response_key: str
+        :param verbose: Provides additional logging if set to True.  Defaults to False.
+        :type verbose: bool
+
+        :return: A List of the combined results of the provided API call.
+        :rtype: List
+        """
+        name_collection = []
+        token = DEFAULT_PAGINATION_TOKEN
+
+        try:
+            while token != "null":
+                response = api_call(nextToken=token)
+                # If response list is not empty, append it to the running list.
+                name_collection += filter(None, response.get(response_key))
+                token = response.get("nextToken")
+
+            self.log.info("Retrieved list of %s %s.", len(name_collection), response_key)
+            if verbose:
+                self.log.info("%s found: %s", response_key.title(), name_collection)
+
+        except ClientError as e:
+            self.log.error(e.response["Error"]["Message"])
+            raise e
+        return name_collection
+
+
+def generate_config_file(
+    eks_cluster_name: str,
+    eks_namespace_name: str,
+    pod_name: str,
+    aws_profile: Optional[str],
+    pod_username: Optional[str] = DEFAULT_POD_USERNAME,
+    pod_context: Optional[str] = DEFAULT_CONTEXT_NAME,
+    aws_region: Optional[str] = None,
+) -> str:
+    """
+    Writes the kubeconfig file given an EKS Cluster.
+
+    :param eks_cluster_name: The name of the cluster to create the EKS Managed Nodegroup in.
+    :type eks_cluster_name: str
+    :param eks_namespace_name: The namespace to run within kubernetes.
+    :type eks_namespace_name: str
+    :param pod_name: The unique name to give the pod.  Used as an identifier in the config filename.
+    :type pod_name: str
+    :param aws_profile: The named profile containing the credentials to use.
+    :type aws_profile: str
+    :param pod_username: The username under which to execute the pod.
+    :type pod_username: str
+    :param pod_context: The name of the context access parameters to use.
+    :type pod_context: str
+    :param aws_region: The name of the AWS Region the EKS Cluster resides in.
+    :type aws_region: str
+    """
+    # Set up the client
+    session = boto3.Session(region_name=aws_region, profile_name=aws_profile)
```

Comment:
we should use the Hook to create a session because this will provide a uniform way to manage credential, which will allow this operator to be used in any environment i.e.  on-premiss, on GCP, on Cloud Composer and other.  Now, rhis operator can only be used in AWS environment.

Reply:
ACK.

Reply:
Should be addressed by https://github.com/apache/airflow/pull/16571/commits/8687b736994d1911011dd8bd2b2903d82ad30a8c